libs/domain/cart/entry/src/entry.component.ts

```diff
@@ -1,170 +1,203 @@
-import { QuantityEventDetail } from '@spryker-oryx/cart/quantity-input';
 import { ContextController } from '@spryker-oryx/core';
-import { ComponentMixin, ContentController } from '@spryker-oryx/experience';
+import { resolve } from '@spryker-oryx/di';
+import { ContentMixin, defaultOptions } from '@spryker-oryx/experience';
 import {
   ProductContext,
   ProductMediaContainerSize,
 } from '@spryker-oryx/product';
+import {
+  NotificationService,
+  PricingService,
+  SemanticLinkType,
+} from '@spryker-oryx/site';
 import { Size } from '@spryker-oryx/ui';
-import { asyncValue, subscribe } from '@spryker-oryx/utilities';
-import { html, TemplateResult } from 'lit';
+import { i18n } from '@spryker-oryx/utilities';
+import { html, LitElement, PropertyValueMap, TemplateResult } from 'lit';
+import { property, state } from 'lit/decorators.js';
 import { when } from 'lit/directives/when.js';
+import { take } from 'rxjs';
+import { QuantityEventDetail } from '../../quantity-input/src';
+import { CartComponentMixin } from '../../src/mixins';
+import { CartEntry } from '../../src/models';
 import {
-  BehaviorSubject,
-  combineLatest,
-  map,
-  merge,
-  Subject,
-  take,
-  tap,
-} from 'rxjs';
-import { CartEntryOptions, REMOVE_EVENT } from './entry.model';
-import { baseStyles, styles } from './styles';
-
-export class CartEntryComponent extends ComponentMixin<CartEntryOptions>() {
-  static styles = [baseStyles, styles];
+  CartEntryAttributes,
+  CartEntryChangeEventDetail,
+  CartEntryOptions,
+  RemoveByQuantity,
+} from './entry.model';
+import { cartEntryStyles } from './styles';
+
+/**
+ * Supports updating the quantity as well as removing the entry entirely.
+ */
+@defaultOptions({
+  removeByQuantity: RemoveByQuantity.ShowBin,
+  enableId: true,
+  enablePreview: true,
+} as CartEntryOptions)
+export class CartEntryComponent
+  extends CartComponentMixin(ContentMixin<CartEntryOptions>(LitElement))
+  implements CartEntryAttributes
+{
+  static styles = [cartEntryStyles];
+
+  @property() key?: string;
```

Comment:
As is mentioned in cart entries, need to find better name

Reply:
see former comment.